game/notegraph.cc

```diff
@@ -179,18 +179,32 @@ void NoteGraph::drawNotes() {
 		  default: throw std::logic_error("Unknown note type: don't know how to render");
 		}
 		double x = m_baseX + it->begin * pixUnit + m_noteUnit; // left x coordinate: begin minus border (side borders -noteUnit wide)
-		double ybeg = m_baseY + (it->notePrev + 1) * m_noteUnit; // top y coordinate (on the one higher note line)
-		double yend = m_baseY + (it->note + 1) * m_noteUnit; // top y coordinate (on the one higher note line)
+		double bar_height = barHeight();
+		double ybeg = m_baseY + (it->notePrev +bar_height) * m_noteUnit; // top y coordinate (on the one higher note line)
+		double yend = m_baseY + (it->note +bar_height) * m_noteUnit; // top y coordinate (on the one higher note line)
 		double w = (it->end - it->begin) * pixUnit - m_noteUnit * 2.0; // width: including borders on both sides
-		double h = -m_noteUnit * 2.0; // height: 0.5 border + 1.0 bar + 0.5 border = 2.0
-		drawNotebar(*t1, x, ybeg, yend, w, h);
+		double h_x = -m_noteUnit * 2.0; // height: 0.5 border + 1.0 bar + 0.5 border = 2.0
+		double h_y = h_x * bar_height; //
+		drawNotebar(*t1, x, ybeg, yend, w, h_x, h_y);
 		if (alpha > 0.0) {
 			ColorTrans c(Color::alpha(alpha));
-			drawNotebar(*t2, x, ybeg, yend, w, h);
+			drawNotebar(*t2, x, ybeg, yend, w, h_x, h_y);
 		}
 	}
 }
 
+double NoteGraph::barHeight() {
+	switch(config["game/difficulty"].i()){
+		case 0:
+			return 1;
+		case 1:
+			return 0.5;
+		case 2:
+			return 0.21;
+	}
+	return 1;
```

Comment:
Add & merge this to the switch case as _default_ case. This gets rid of the `case 0:` block